### PR TITLE
Add an official Governance structure

### PR DESCRIPTION
**Pull request by odow**
This PR adds a formal governance structure for the JuMP project. 

A related document is the SciML governance page: https://sciml.ai/governance/

Before merging, this PR requires approval from:

- [x] @mlubin
- [x] @blegat
- [x] @joaquimg 
- [x] @joehuchette 
- [x] @odow
- [x] @juan-pablo-vielma
- [x] @chkwon
- [x] @iainnz (optional)
- [x] Consensus on monthly developer call

### CODE REVIEW

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:59]: shall disclose to whom? to the public or to the other members of the council?

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:49]: steering committee or steering council?

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:61]: What does the council vote on? (Not described)

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:79]: This discussion about discourse doesn't really fit in with the governance structure.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:86]: Also doesn't fit in with the governance structure, unless it's a statement like "the council will maintain a developer chat room ...".

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:101]: For a governance doc, this should talk about how the workshops are organized, e.g., the council chooses a chair.

**Review by mlubin — COMMENTED** [review]
It's good that we have a starting point now. But this is missing a discussion about how decisions (technical and otherwise) are ultimately made. BDFL or majority vote or consensus? May need to discuss in a call.

**Comment by odow** [comment]
> But this is missing a discussion about how decisions (technical and otherwise) are ultimately made. BDFL or majority vote or consensus?

Consensus discussion on Github or a monthly developer calls. But we nominate you as BDFL with sole veto power? Agree probably needs to be discussed on call.

File: pages/governance.md
Comment (odow) [reply pages/governance.md]: Just to the council?

File: pages/governance.md
Comment (odow) [reply pages/governance.md]: Changed to `recuse themselves from authorizing or approving any expenditures.`. It's really just the financial COI.

**Comment by mlubin** [comment]
(Not yet commenting on the BDFL nomination.) Another important issue is allocation of commit bits and having a clear pathway for community members to become either part of the core team or an effective owner of a JuMP-dev repository (e.g., wrappers).

File: pages/governance.md
Comment (odow) [inline pages/governance.md:151]: This is a good reason for standardization. I took a look and the outside collaborators have a mix of `Write`, `Maintain`, and `Admin` rights.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:172]: I would say that a new core contributor should be added by consensus of the current core contributors and notification to the steering council. I don't think the steering council needs to get involved in technical decisions unless they're strategically significant.

File: pages/governance.md
Comment (odow) [reply pages/governance.md]: Here's the breakdown: https://docs.github.com/en/free-pro-team@latest/github/setting-up-and-managing-organizations-and-teams/repository-permission-levels-for-an-organization#repository-access-for-each-permission-level

The short answer is yes.

The main difference between Maintain and Admin is the ability to edit settings (archive, default branches, change deploy keys), invite external collaborators, delete issues, etc.

I think Maintain is the one we want. We don't want external collaborators adding new collaborators, or deleting the repository, for example. 

For people transferring repositories, there is a trade-off: transfer to JuMP-dev and get structure, help with maintenance, etc, and loose some permissions such as the ability to add collaborators.

File: pages/governance.md
Comment (odow) [inline pages/governance.md:151]: Is there a reason @joehuchette has `Member` permission in JuMP-dev and not `Owner`?

Comment (mlubin) [reply pages/governance.md:151]: A related question: @joehuchette, would you prefer to be listed under core contributors or emeritus core contributors?

File: pages/governance.md
Comment (joehuchette) [inline pages/governance.md:143]: *received

File: pages/governance.md
Comment (joehuchette) [inline pages/governance.md:165]: *decisions are

File: pages/governance.md
Comment (joehuchette) [reply pages/governance.md:151]: A bit conflicted, mostly because "emeritus" suggests retiring from contributing in perpetuity, which is not my intention. I'll defer to the BDFL :)

**Review by mlubin — COMMENTED** [review]
I would suggest a bit of a restructuring to clarify the roles of the core developers and steering council. Right now, core developers are a level 3 heading under "commit rights" while the steering council is a level 2 heading, which doesn't seem right. The distinction between what's a technical decision that falls to core developers and what's "other" that falls to the steering council is also ambiguous.

What do you think?

In my view, the steering council should be secondary to the core developers. In terms of decision making, the core developers have ultimate authority on the direction of the project (nothing can really happen without their agreement). The steering council is there to support the core developers, e.g., by handling finances and representing JuMP with NumFOCUS. The steering council can gain additional decision-making power if the core developers decide to delegate, e.g., choosing a chair for the JuMP-dev workshop.

**Comment by odow** [comment]
> What do you think?

Agree. How's this now?

**Review by mlubin — COMMENTED** [review]
I have more comments on the text, but first a larger refactoring suggestion. The decision-making process discussion is smeared across different sections, e.g., "Core contributors are the ultimate authority on the direction of the JuMP project." in the definition of core contributors.

What about:
1) Basic definition of the various entities, i.e., core contributors, emeritus core contributors, repository maintainers, steering council (interface with NumFOCUS), BDFL, and how membership in each group is determined.
2) The decision-making process
3) GitHub permissions for each group (if applicable). The GitHub permissions are there to enable people to carry out the decision-making process, so they're secondary in that respect.
4) Channels of community involvement

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:112]: I would move "Conflict of interest" under the "decision making process" section and extend it to all decisions made by the steering council or core contributors. Core contributors should also disclose conflicts of interest with other core contributors and step back from decisions when conflicts of interests are in play.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:140]: This sentence is related to non-fiscal decisions.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:134]: I think we mean financial more than fiscal.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:138]: Nit: "acting" is redundant.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:168]: This is a setting that's enforced by GitHub, not sure how important it is to mention here. I would say that GitHub permissions are used to control access to repositories.

Also: Anyone with commit access to a repository is trusted to use it in a way that's consistent with the "non-fiscal decisions" process. Prefer pull requests over direct pushes, ask for feedback on changes if you're not sure there's a consensus, and follow JuMP's style guide and development processes.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:179]: "Emeritus core contributors retain the same commit rights as core contributors, unless they choose to surrender them."

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:199]: The steering council isn't doing much to maintain the forum. Do we need to assign this task to them, or just say that *there is* a community form? Same for the chat room.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:223]: We can just choose to delegate this to the steering council permanently (don't see why we'd want to change it).

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:56]: "... who were core contributors, but have stepped back to a less active role. A core developer may choose to switch to emeritus status by informing the other core developers and the steering council."

File: pages/governance.md
Comment (odow) [inline pages/governance.md:69]: Do we need something about the core contributors may remove someone as a maintainer if they abuse their access, or if they are inactive for a period of time?

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:180]: For a consistent tone, "Those with permissions should prefer pull requests ..."

File: pages/governance.md
Comment (mlubin) [reply pages/governance.md:69]: I'd say, "Repository maintainers can step back from their role at any time by informing a core developer. Furthermore, core developers. by consensus. can choose to remove repository maintainers for any reason (typically inactivity)."

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:184]: I don't think this threatening sentence is needed. From my suggestion above, core developers would be able to remove repository maintainers for any reason, which would include abuse. More extreme cases are also covered by the code of conduct and conflict of interest sections.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:271]: Maybe this is a good place to mention that we request you not use "JuMP" in the name of third-party packages.

File: pages/governance.md
Comment (mlubin) [reply pages/governance.md]: The only mechanism defined for transitioning from being a core developer to an emeritus core developer is by request, so you'll remain a core developer until you ask not to be :)

**Comment by mlubin** [comment]
Now's a good time to get feedback from the other core developers. @blegat @joaquimg

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:83]: This might create an artificial barrier. Some people might be good repository maintainers but not want to have their name on the README, which potentially invites attention and demands from users. We could use a separate and less user-facing page to list repository maintainers.

Comment (odow) [reply pages/governance.md]: Fair point. I've removed for now. We can revisit if/how to give maintainers recognition in future. Are you thinking a `MAINTAINERS.md` in each repo? Or a page on `jump.dev`?

Comment (mlubin) [reply pages/governance.md]: I was thinking a page on jump.dev.

File: pages/governance.md
Comment (joehuchette) [inline pages/governance.md:12]: I would propose "powerful" and "easy-to-use" as other potential descriptors.

File: pages/governance.md
Comment (joehuchette) [inline pages/governance.md:29]: Worth spelling out what this is an acronym for at least once, somewhere.

File: pages/governance.md
Comment (joehuchette) [inline pages/governance.md:51]: several repositories under the jump-dev umbrella?

File: pages/governance.md
Comment (joaquimg) [inline pages/governance.md:136]: remove space

File: pages/governance.md
Comment (joaquimg) [inline pages/governance.md:130]: make it clearer how proposals are tabled, same for non-financial.
possibly this additional detail could live in: ### Monthly Developer Call
and could be as simple as ping mlubin or odow

File: pages/governance.md
Comment (joaquimg) [inline pages/governance.md:246]: Should we include jump's twitter here?

**Review by joaquimg — APPROVED** [review]
Great work!
Left some comments.

File: pages/governance.md
Comment (odow) [reply pages/governance.md:22]: Once this governance document is finalized, I propose we replace the `CODE_OF_CONDUCT.md` with a link here instead.

File: _posts/2020-10-20-governance.md
Comment (blegat) [inline _posts/2020-10-20-governance.md:36]: belive -> believe

**Review by blegat — APPROVED** [review]
Looks good!

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:96]: The name of the list is jump-steering-committee@...
Did we intend to rename the committee to a council?

Comment (odow) [reply pages/governance.md:96]: > Did we intend to rename the committee to a council?

I copied the SciML language, and couldn't remember if it was a council or a committee. Will change.

File: pages/governance.md
Comment (chkwon) [inline pages/governance.md:147]: such as

File: pages/governance.md
Comment (chkwon) [inline pages/governance.md:154]: additional

File: pages/governance.md
Comment (chkwon) [inline pages/governance.md:270]: do not use

File: pages/governance.md
Comment (chkwon) [inline pages/governance.md:275]: maintenance

**Review by chkwon — COMMENTED** [review]
Looks good to me.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:184]: It's fine to say "consistent with the Decision Making Process", because commit access might be used for financial-related decisions also.

File: pages/governance.md
Comment (mlubin) [inline pages/governance.md:32]: Nit: "holds the title of" sounds better

Comment (odow) [reply pages/governance.md]: Fixed, boss.

**Review by joehuchette — APPROVED** [review]
LGTM

**Review by mlubin — APPROVED** [review]

**Review by chkwon — APPROVED** [review]

**Comment by juan-pablo-vielma** [comment]
LGTM

**Review by juan-pablo-vielma — APPROVED** [review]

**Comment by IainNZ** [comment]
LGTM

**Comment by odow** [comment]
I'll leave this open for the weekend if anyone has additional comments, particularly the new wording of the mission. If there are no issues raised, I will merge Sunday evening US Eastern time.

**Comment by joaquimg** [comment]
One minor, non-blocking, comment.
Did we put the governance tab as the second tab on the webpage on purpose?

In https://sciml.ai/governance/ it is close to the end...
In https://scikit-learn.org/stable/about.html is it even a bit hidden in "about us".
A bit hidden in https://docs.scipy.org/doc/scipy/reference/index.html also...

Not suggesting it should be hidden, but I would move it to the end.

**Comment by odow** [comment]
We now have:
image
I moved the forum, chatroom, and Code/Ecosystem page under "Community", and renamed "News" to "Blog"

File: _config.yml
Comment (mlubin) [inline _config.yml:120]: This is a developer chatroom. We shouldn't be funneling users into it by advertising it as the JuMP chatroom.

**Pull request merged**
